spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestCreateActions.java

```diff
@@ -162,6 +162,9 @@ public void before() {
     this.tableLocation = tableDir.toURI().toString();
     this.catalog = (TableCatalog) spark.sessionState().catalogManager().catalog(catalogName);
 
+    if (!catalogName.equals("spark_catalog")) {
+      spark.conf().set("spark.sql.catalog.spark_catalog", SparkSessionCatalog.class.getName());
```

Comment:
I don't think this ever needs to be set?

Reply:
This is needed to migrate from a non-Iceberg catalog table.

Reply:
sorry, this is probably my bad due to my earlier suggestion. I think we can remove setting this here if we remove `spark.conf().unset("spark.sql.catalog.spark_catalog");`. That way all tests should still pass